packages/components/file-card/src/file-card.js

```diff
@@ -0,0 +1,357 @@
+import { TSElement, unsafeCSS, html } from '@tradeshift/elements';
+import '@tradeshift/elements.typography';
+import '@tradeshift/elements.card';
+import '@tradeshift/elements.progress-bar';
+import '@tradeshift/elements.file-size';
+import css from './file-card.css';
+
+customElements.define(
+	'ts-file-card',
+	class extends TSElement {
+		constructor() {
+			super();
+			this.state = 'upload';
+			this.size = 'full';
+			this.removable = false;
+			this.toggleRemoveActionHoverState = this.toggleRemoveActionHoverState.bind(
+				this
+			);
+			this.toggleDownloadActionHoverState = this.toggleDownloadActionHoverState.bind(
+				this
+			);
+			this.removeFile = this.removeFile.bind(this);
+			this.downloadFile = this.downloadFile.bind(this);
+		}
+
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				state: { type: String, reflect: true },
+				fileObject: { type: Object, attribute: 'file-object' },
+				rtl: { type: Boolean, reflect: true },
+				removable: { type: Boolean, reflect: true },
+				size: { type: String, reflect: true },
+				errorMessage: {
+					type: String,
+					reflect: true,
+					attribute: 'error-message'
+				}
+			};
+		}
+
+		get cardType() {
+			const cardTypes = {
+				uploading: 'focus',
+				download: 'default',
+				failed: 'failed'
+			};
+			return cardTypes[this.state];
+		}
+
+		get errorMessages() {
+			if (this.state === 'failed') {
+				const message = this.errorMessage;
+				return html`
+					<ts-typography
+						class="error-message"
+						text="${message}"
+						color="error"
+						variant="subtitle"
+						no-wrap
+					></ts-typography>
+				`;
+			}
+		}
+
+		get fileIcon() {
+			return html`
+				<span class="file-icon-wrapper icon-wrapper">
+					<!--TODO: use icon components-->
+					<svg
+						class="file-icon"
+						width="32px"
+						height="42px"
+						viewBox="0 0 32 42"
+						version="1.1"
+						xmlns="http://www.w3.org/2000/svg"
+						xmlns:xlink="http://www.w3.org/1999/xlink"
+					>
+						<g id="Symbols" stroke-width="1">
+							<g
+								id="card/5.-File-card/mouseover"
+								transform="translate(-9.000000, -9.000000)"
+							>
+								<g id="5.-File-card">
+									<g id="File-icon" transform="translate(10.000000, 10.000000)">
+										<path
+											d="M19.5667631,-0.5 L29.5322581,9.79767818 L29.5322581,38.0952381 C29.5322581,39.3654698 28.5111819,40.5 27.3244782,40.5 L1.70777989,40.5 C0.521076133,40.5 -0.5,39.3654698 -0.5,38.0952381 L-0.5,1.9047619 C-0.5,0.634530207 0.521076133,-0.5 1.70777989,-0.5 L19.5667631,-0.5 Z"
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 285 characters long line (which is greater than 120 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3ALineLength)